impl/src/main/kotlin/nl/nuts/consent/bridge/rpc/CordaRPClientFactory.kt

```diff
@@ -117,44 +133,58 @@ class CordaRPClientWrapper : AutoCloseable {
-                val nodeInfo = unvalidatedConnection.proxy.nodeInfo()
-                require(nodeInfo.legalIdentitiesAndCerts.isNotEmpty())
-                unvalidatedConnection
-            } catch(secEx: ActiveMQSecurityException) {
-                // Incorrect credentials, log and rethrow
-                logger.error(secEx.message)
-                throw secEx
-            } catch(ex: RPCException) {
-                // Deliberately not logging full stack trace as it will be full of internal stacktraces.
-                logger.error("Exception upon establishing connection: " + ex.message)
-                retryCount--
-                if (retryCount == 0) {
-                    shutdown = true
+            try {
+                // first try socket, because RPCCLient will hang indefinitely
+                val clientSocket = Socket()
+                clientSocket.connect(InetSocketAddress(consentBridgeRPCProperties.host, consentBridgeRPCProperties.port), retryInterval.toMillis().toInt())
+                clientSocket.close()
+
+                connection = try {
+                    logger.info("Connecting to: $nodeAddress")
+
+                    val gracefulReconnect = GracefulReconnect(
+                            maxAttempts = consentBridgeRPCProperties.retryCount,
+                            onDisconnect = { logger.error("disconnected from RPC") },
+                            onReconnect = { logger.info("RPC connection re-established") }
+                    )
+
+                    val client = CordaRPCClient(
+                            nodeAddress,
+                            object : CordaRPCClientConfiguration() {
+                                override val connectionMaxRetryInterval = retryInterval
+                                override val maxReconnectAttempts = consentBridgeRPCProperties.retryCount
+                            }
+                    )
+                    val unvalidatedConnection = client.start(
+                            username = consentBridgeRPCProperties.user,
+                            password = consentBridgeRPCProperties.password,
+                            gracefulReconnect = gracefulReconnect
+                    )
+
+                    // Check connection is truly operational before returning it.
+                    val nodeInfo = unvalidatedConnection.proxy.nodeInfo()
+                    require(nodeInfo.legalIdentitiesAndCerts.isNotEmpty())
+                    unvalidatedConnection
+                } catch (secEx: ActiveMQSecurityException) {
+                    // Incorrect credentials, log and rethrow
+                    logger.error(secEx.message)
+                    throw secEx
+                } catch (ex: RPCException) {
+                    // Deliberately not logging full stack trace as it will be full of internal stacktraces.
+                    logger.error("Exception upon establishing connection: " + ex.message)
```

Comment:
you could use Kotlin string interpolation here

Reply:
switched to slf4j parameter